tb-scripts/download_apis.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+import os
+import shutil
+from zipfile import ZipFile
+import requests
+
+
+def download_tip(url, filename):
+    print(f"Downloading {url} as {filename}, might take a while")
+
+    with requests.get(url, stream=True) as r:
+        with open(filename, "wb") as fh:
+            for chunk in r.iter_content(8192):
+                if chunk:
+                    fh.write(chunk)
+
+
+def main(working_dir, existing_source_dir, mozilla_central_zip, comm_central_zip):
+    archives = [
+        (mozilla_central_zip, "mozilla-central"),
+        (comm_central_zip, "comm-central")
+    ]
+
+    mozilla_central_folder = None
+    comm_central_folder = None
+    merged_folder = "thunderbird_api"
+    base_dir = os.path.join('./', working_dir, 'addons_linter')
+
+    if existing_source_dir is not None:
+        mozilla_central_folder = existing_source_dir
+        comm_central_folder = os.path.join(existing_source_dir, "comm")
+    else:
+        for archive in archives:
+            filename = os.path.join(working_dir, f"{archive[1]}.zip")
+            if not os.path.isfile(filename):
+                download_tip(archive[0], filename)
+
+        # Mozilla-Central
+        print("Extracting schemas from mozilla-central.zip")
+        with ZipFile(os.path.join(working_dir, "mozilla-central.zip"), "r") as mozilla_central:
+            name_list = mozilla_central.namelist()
+            extract_list = []
+
+            # Grab the folder name
+            mozilla_central_folder = name_list[0].split('/')[0]
+
+            for file in name_list:
+                if "toolkit/components/extensions/schemas" in file \
+                        or "browser/components/extensions/schemas" in file:
+                    extract_list.append(file)
+
+            mozilla_central.extractall(base_dir, extract_list)
+
+        # Comm-Central
+        print("Extracting schemas from comm-central.zip")
+        with ZipFile(os.path.join(working_dir, "comm-central.zip"), "r") as comm_central:
+            name_list = comm_central.namelist()
+            extract_list = []
+
+            # Grab the folder name
+            comm_central_folder = name_list[0].split('/')[0]
+
+            for file in name_list:
+                if "mail/components/extensions" in file:
+                    extract_list.append(file)
+
+            comm_central.extractall(base_dir, extract_list)
+
+    # Copying schemas over
+    print("Copying and merging schemas")
+
+    files = {
+        mozilla_central_folder: [
+            'toolkit/',
+            'browser/components/extensions/schemas/commands.json',
```

Comment:
We have our own copy of `commands.json` and it should not be merged with the firefox version.

However, according to this file:
https://searchfox.org/comm-central/source/mail/components/extensions/jar.mn#58

we do clone `pkcs11.json` which is unusual, but I guess the rationale is to not have to monitor upstream for changes. But having that hardcoded here may get forgotten later. I will ask rob, if we can copy that file to c-c during built-time. We will have to keep including it here for now.